mambaSharedFramework/HLSValueTypes.swift

```diff
@@ -266,4 +454,68 @@ public func ==(lhs: HLSCodecArray, rhs: HLSCodecArray) -> Bool {
     return lhs.codecs == rhs.codecs
 }
 
+/// Represents information to assist in view presentation.
+///
+/// Indicates when video content in the Variant Stream requires specialized rendering to be properly displayed.
+public struct HLSVideoLayout: Equatable, FailableStringLiteralConvertible {
+    /// Each specifier controls one aspect of the entry. That is, the specifiers are disjoint and the values for a
+    /// specifier are mutually exclusive.
+    public let layouts: [VideoLayout]
+    /// The client SHOULD assume that the order of entries reflects the most common presentation in the content.
+    ///
+    /// For example, if the content is predominantly stereoscopic, with some brief sections that are monoscopic then the
+    /// Multivariant Playlist SHOULD specify `REQ-VIDEO-LAYOUT="CH-STEREO,CH-MONO"`. On the other hand, if the content
+    /// is predominantly monoscopic then the Multivariant Playlist SHOULD specify `REQ-VIDEO-LAYOUT="CH-MONO,CH-STEREO"`.
+    public let predominantLayout: VideoLayout
+
+    public enum VideoLayout: String {
+        /// Monoscopic.
+        ///
+        /// Indicates that a single image is present.
+        case chMono = "CH-MONO"
+        /// Stereoscopic.
+        ///
+        /// Indicates that both left and right eye images are present.
+        case chStereo = "CH-STEREO"
+
+        init?(str: Substring) {
+            switch str {
+            case "CH-MONO": self = .chMono
+            case "CH-STEREO": self = .chStereo
+            default: return nil
+            }
+        }
+    }
+
+    public init?(string: String) {
+        var layouts = [VideoLayout]()
+        for str in string.split(separator: ",") {
+            if let layout = VideoLayout(str: str) {
+                layouts.append(layout)
+            } else {
+                // Favor failing to parse the whole array if we find an unrecognized layout, so that we don't risk mis-
```

Comment:
Confused on this-

If there is an unrecognized layout, you want to toss the entire `HLSVideoLayout` type out and return nil?

Or did you mean to just skip the unrecognized one and keep parsing?

Reply:
I mean toss the whole thing out... This is similar to my approach for special usage identifiers in the CHANNELS above. Same as there, I could get around this, by adding an `unknown(String)` case... What do you think?

Reply:
I added an `unrecognized(String)` case here: 02d2cb9f8a36379b6b1d49891da3bae364c1309c